dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/h2/CancelStreamException.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.remoting.http12.h2;
+
+import org.apache.dubbo.remoting.http12.ErrorCodeHolder;
+
+public class CancelStreamException extends RuntimeException implements ErrorCodeHolder {
+
+    private final boolean cancelByRemote;
+
+    private final long errorCode;
+
+    private CancelStreamException(boolean cancelByRemote, long errorCode) {
+        this.cancelByRemote = cancelByRemote;
+        this.errorCode = errorCode;
+    }
+
+    public boolean isCancelByRemote() {
+        return cancelByRemote;
+    }
+
+    public static CancelStreamException fromRemote(long errorCode) {
+        return new CancelStreamException(true, errorCode);
+    }
+
+    public static CancelStreamException fromLocal(long errorCode) {
+        return new CancelStreamException(false, errorCode);
+    }
+
+    @Override
+    public long getErrorCode() {
+        return errorCode;
+    }
+}
```

Comment:
**Security Issue**: The class CancelStreamException exposes a public static factory method for creating instances that signify a cancellation by remote entities without any form of authentication or verification of the remote entity's identity, potentially allowing an attacker to forge such exceptions to disrupt the service. <br> **Fix**: Introduce a mechanism to authenticate or verify the legitimacy of the remote entity before allowing the creation of a CancelStreamException instance that signifies a remote cancellation. This could involve checking a signature, a shared secret, or a secure context. <br> **Code Suggestion**: 
 ```
 private static final Object sharedSecret = "SECRET";

 public static CancelStreamException fromRemote(long errorCode, Object secret) {
     if (!sharedSecret.equals(secret)) {
         throw new SecurityException("Invalid secret for remote cancellation.");
     }
     return new CancelStreamException(true, errorCode);
 }
 ```